### PR TITLE
Update-DeepFloyd-IF-Pipelines-Docstrings

### LINKED ISSUE #5302: [Hacktoberfest] DeepFloyd IF docstrings
This feature request is included in this year's Diffusers Hacktoberfest 🎃!

Hacktoberfest's purpose is to give upcoming open-source developers the opportunity to learn about open-source and programming by fixing simple issues (like this one).

How to solve the issue:

- Every issue has a description further below explaining the problem. First, you should try to understand the issue.
- Second, follow the [contribution guide](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md) to learn how to contribute to Diffusers.
- Claim this issue for you by opening a PR that links to this issue and writing, "I am solving this issue in <link-to-pr>".

Issue

For each of the DeepFloyd IF pipelines:

1. The [`encode_prompt`](https://huggingface.co/docs/diffusers/main/en/api/pipelines/deepfloyd_if#diffusers.IFPipeline.encode_prompt) parameters are all mixed up and need to be reformatted
2. The parameter values in [`__call__`](https://huggingface.co/docs/diffusers/main/en/api/pipelines/deepfloyd_if#diffusers.IFPipeline.__call__) are out of sync (for example, `num_inference_steps` should be 100 but the docstring says it is 50). Please also check if other values are also incorrect, like the `guidance_scale`

### PR DESCRIPTION
**Pull request by m0saan**
# What does this PR do?

Fixes #5302

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by DN6** [comment]
@m0saan Not sure what the change is here?

**Comment by m0saan** [comment]
@DN6 I am still working on it locally, I did that to claim the issue first :p

**Comment by m0saan** [comment]
@stevhliu @DN6 can you review please, Thanks!

File: src/diffusers/pipelines/deepfloyd_if/pipeline_if_img2img.py [hunk 1]
```diff
@@ -184,41 +184,42 @@ def remove_all_hooks(self):
...
                 Pre-generated text embeddings. Can be used to easily tweak text inputs, *e.g.* prompt weighting. If not
                 provided, text embeddings will be generated from `prompt` input argument.
             negative_prompt_embeds (`torch.FloatTensor`, *optional*):
                 Pre-generated negative text embeddings. Can be used to easily tweak text inputs, *e.g.* prompt
                 weighting. If not provided, negative_prompt_embeds will be generated from `negative_prompt` input
                 argument.
+            do_classifier_free_guidance (`bool`, *optional*, defaults to `True`):
+                whether to use classifier free guidance or not
+            clean_caption (bool, defaults to False):
+                If True, the function will preprocess and clean the provided caption before encoding.
```
Comment (stevhliu) [inline src/diffusers/pipelines/deepfloyd_if/pipeline_if_img2img.py:220]:
```suggestion
            clean_caption (bool, defaults to `False`):
                If `True`, the function will preprocess and clean the provided caption before encoding.
```

File: src/diffusers/pipelines/deepfloyd_if/pipeline_if_superresolution.py [hunk 2]
```diff
@@ -637,19 +639,19 @@ def __call__(
             prompt (`str` or `List[str]`, *optional*):
                 The prompt or prompts to guide the image generation. If not defined, one has to pass `prompt_embeds`.
                 instead.
-            height (`int`, *optional*, defaults to self.unet.config.sample_size):
+            height (`int`, *optional*, defaults to None):
```
Comment (stevhliu) [inline src/diffusers/pipelines/deepfloyd_if/pipeline_if_superresolution.py:642]: Good catch!

**Review by stevhliu — APPROVED** [review]
Great job, thanks for the fixes! :)

File: src/diffusers/pipelines/deepfloyd_if/pipeline_if.py [hunk 3]
```diff
@@ -161,38 +161,40 @@ def remove_all_hooks(self):
     @torch.no_grad()
     def encode_prompt(
         self,
-        prompt,
-        do_classifier_free_guidance=True,
-        num_images_per_prompt=1,
-        device=None,
-        negative_prompt=None,
+        prompt: Union[str, List[str]],
```
Comment (patrickvonplaten) [inline src/diffusers/pipelines/deepfloyd_if/pipeline_if.py:164]: This we need to keep the ordering here to not break backwards compatibility

**Comment by patrickvonplaten** [comment]
Great job @m0saan !

**Pull request merged**